front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsControllerV2Tck.groovy

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.netflix.spinnaker.front50.controllers.v2
+
+import com.amazonaws.ClientConfiguration
+import com.amazonaws.services.s3.AmazonS3Client
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.front50.exception.InvalidApplicationRequestException
+import com.netflix.spinnaker.front50.exception.NotFoundException
+import com.netflix.spinnaker.front50.model.S3ApplicationDAO
+import com.netflix.spinnaker.front50.model.application.Application
+import com.netflix.spinnaker.front50.model.application.ApplicationDAO
+import com.netflix.spinnaker.front50.model.application.CassandraApplicationDAO
+import com.netflix.spinnaker.front50.model.notification.NotificationDAO
+import com.netflix.spinnaker.front50.model.pipeline.PipelineDAO
+import com.netflix.spinnaker.front50.model.pipeline.PipelineStrategyDAO
+import com.netflix.spinnaker.front50.model.project.ProjectDAO
+import com.netflix.spinnaker.front50.utils.CassandraTestHelper
+import com.netflix.spinnaker.front50.utils.S3TestHelper
+import com.netflix.spinnaker.front50.validator.HasEmailValidator
+import com.netflix.spinnaker.front50.validator.HasNameValidator
+import org.springframework.context.support.StaticMessageSource
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+import rx.schedulers.Schedulers
+import spock.lang.IgnoreIf
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Subject
+
+import java.util.concurrent.Executors
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+abstract class ApplicationsControllerV2Tck extends Specification {
```

Comment:
Looking at the history, I originally borrowed the term from `orca` (where we also have a couple *Tck) as these controller tests are theoretically able to verify behavior against both `s3` and `cassandra`.

I have loose allegiance to `*Tck` and wouldn't be bent out of shape if we changed the suffix.